server/utils/validation.ts

```diff
@@ -18,9 +18,14 @@ export const catchValidationErrorOrPropogate = (
   redirectPath: string,
 ): void => {
   if ('data' in error) {
-    const invalidParams = error.data['invalid-params'] || error.data
+    let errors: Record<string, string>
```

Comment:
Up to you but I think we could do:
`const errors = error.data['invalid-params'] ? generateErrors(error.data['invalid-params']) : error.data as Record<string, string>`

Reply:
Oooh, that's better. Big fan of one-liners